src/pyhf/optimize/opt_tflow.py

```diff
@@ -46,7 +46,7 @@ def func(pars):
             with tf.GradientTape() as tape:
                 tape.watch(pars)
                 constrained_pars = tv.stitch([fixed_values_tensor, pars])
-                constr_nll = objective(constrained_pars, data, pdf)
+                constr_nll = objective(constrained_pars, data, pdf)[0]
             grad = tape.gradient(constr_nll, pars).values
             return constr_nll.numpy(), grad
```

Comment:
this one should be 
```suggestion
            return constr_nll.numpy()[0], grad
```

rather than the `[0]` above I think. I'd check if there's a difference...

Reply:
I think you're right, as otherwise you're converting early. I copied what @lukasheinrich's diff showed, but I'll push with this change and we can revert if needed.

Reply:
I think either way is fine